lib/net/ldap/connection.rb

```diff
@@ -86,7 +87,26 @@ def self.wrap_with_ssl(io, tls_options = {})
     ctx.set_params(tls_options) unless tls_options.empty?
 
     conn = OpenSSL::SSL::SSLSocket.new(io, ctx)
-    conn.connect
+
+    begin
+      if timeout
+        conn.connect_nonblock
+      else
+        conn.connect
+      end
+    rescue IO::WaitReadable
+      if IO.select([conn], nil, nil, timeout)
+        retry
+      else
+        raise Net::LDAP::LdapError, "OpenSSL connection read timeout"
```

Comment:
`Errorno::ETIMEDOUT` makes more sense in terms of compatibility since `TCPSocket` raises `Errorno::ETIMEDOUT`. 

@jch @mtodd how do you think?